### PR TITLE
feat(console): add printer for mapiterator

### PR DESCRIPTION
**Pull request by liz3**
MapIterator was not supported in printing, libraries like discordjs make big use of maps and so i think supporting them would be a good idea.

### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
- Manual test

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
This looks good, but needs a test. You can use `Bun.inspect` to write the test.

File: src/bun.js/bindings/exports.zig [hunk 1]
```diff
@@ -1532,6 +1534,7 @@ pub const ZigConsoleClient = struct {
                         JSValue.JSType.BooleanObject => .Boolean,
                         JSValue.JSType.JSFunction => .Function,
                         JSValue.JSType.JSWeakMap, JSValue.JSType.JSMap => .Map,
+                        JSValue.JSType.JSMapIterator => .MapIterator,
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/exports.zig:1539]: Do we need to worry about `SetIterator`?

Comment (liz3) [reply src/bun.js/bindings/exports.zig:1539]: il double check

**Comment by liz3** [comment]
@Jarred-Sumner i added a printer for `SetIterator` and added tests :)

**Review by Jarred-Sumner — APPROVED** [review]
Thank you

**Pull request merged**